backend/src/core/services/campaign.service.ts

```diff
@@ -144,25 +150,95 @@ const listCampaigns = ({
   userId,
   offset,
   limit,
+  type,
+  status,
+  name,
+  sortBy,
+  orderBy,
 }: {
   userId: number
   offset: number
   limit: number
+  type?: ChannelType
+  status?: Status
+  name?: string
+  sortBy?: SortField
+  orderBy?: Ordering
 }): Promise<{ rows: Array<Campaign>; count: number }> => {
   const campaignJobs = '(PARTITION BY "job_queue"."campaign_id")'
   const maxAge = config.get('redaction.maxAge')
 
+  let whereFilter: {
+    user_id: number
+    type?: ChannelType
+  } = { user_id: userId }
+
+  if (type) {
+    whereFilter.type = type
+  }
+
+  if (name) {
+    const nameMatch = {
+      name: {
+        [Op.iLike]: '%' + name + '%',
```

Comment:
Just wanna remind here to double check for potential sql injections

Reply:
Rae talked to Preston; apparently Sequelize parametrize the query, so there shouldn't be injection points